### DISCUSSION TITLE
Preferences table missing user_group_id column

### DISCUSSION THREAD

**Discussion by Jack-XHP** [firefly-iii/firefly-iii#8815] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

The new version pushed to docker hub causes this issue and will lead to a 500 internal server error due to existing db doesn't have user_group_id column. Manually adding that column to preferences table via SQL fix this issue

### Debug information

Unfortunately, this error was not recoverable :(. Firefly III broke. The error is:

SQLSTATE[42S22]: Column not found: 1054 Unknown column 'user_group_id' in 'where clause' (Connection: mysql, SQL: select `id`, `user_id`, `name`, `data`, `updated_at`, `created_at` from `preferences` where `user_group_id` is null and `user_id` = 2 and `name` = viewRange limit 1)

This error occured in file /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php on line 813 with code 42S22.

**Comment by JC5** [comment]
Did you run the migrations?

**Reply by HawkFire76** [reply]
Hello @JC5, I got the same error when upgrading (docker) from 6.1.6 to 6.1.18

[fireflyiii_docker_upgrade_6.1.6_to_6.1.18_user_group_id_missing.log](https://github.com/user-attachments/files/16133324/fireflyiii_docker_upgrade_6.1.6_to_6.1.18_user_group_id_missing.log)
